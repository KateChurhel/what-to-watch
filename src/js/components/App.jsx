// libraries
import React from 'react';
import { Route, Switch } from 'react-router-dom';
// constants
import ROUTES from '../constants/routes';
// HOC
import withCatalogRequest from '../hoc/withCatalogRequest';
// views
import CatalogPage from './Pages/CatalogPage';
import DetailsPage from './Pages/DetailsPage';
import MainPage from './Pages/MainPage';
import PageNotFound from './Pages/404';
import Footer from './Partials/Footer';
import Header from './Partials/Header';

const App = () => (
  <>
    <Header />
    <main>
      <Switch>
        <Route component={withCatalogRequest(DetailsPage)} path={ROUTES.detailsPage} />
        <Route component={withCatalogRequest(CatalogPage)} path={[ROUTES.search, ROUTES.catalog]} />
        <Route
          component={withCatalogRequest(MainPage)}
          exact
          path={ROUTES.mainPage}
        />
        <Route
          component={PageNotFound}
          exact
          path="*"
        />
      </Switch>
    </main>
    <Footer />
  </>
);

export default App;
