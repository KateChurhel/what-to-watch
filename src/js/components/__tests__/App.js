import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import App from '../App';

it('App correctly renders after relaunch', () => {
  const tree = shallow(<App />);
  expect(toJSON(tree)).toMatchSnapshot();
});
