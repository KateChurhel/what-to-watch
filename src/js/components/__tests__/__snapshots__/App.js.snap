// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App correctly renders after relaunch 1`] = `
<Fragment>
  <Header />
  <main>
    <Switch>
      <Route
        component={[Function]}
        path="/details/:category/:cardId"
      />
      <Route
        component={[Function]}
        path={
          Array [
            "/search/:category/:page",
            "/:category/:page",
          ]
        }
      />
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        exact={true}
        path="*"
      />
    </Switch>
  </main>
  <Footer />
</Fragment>
`;
